rust/kernel/net/device/flags.rs

```diff
@@ -0,0 +1,277 @@
+use crate::{bindings, Error};
+use core::convert::{From, TryFrom};
+use core::ops::{Add, AddAssign, Sub, SubAssign};
+
+/// Holds multiple flags to give to an interface via [`super::NetDevice::add_flag`].
+#[repr(transparent)]
+#[derive(Debug, Clone, Copy)]
+pub struct Features(u64);
+
+impl Features {
+    /// Create new Flag with value `0`.
+    #[inline]
+    pub const fn empty() -> Self {
+        Self(0)
+    }
+
+    /// Add flag to Self.
+    #[inline]
+    pub fn insert(&mut self, flag: u64) {
+        self.0 |= flag;
+    }
+
+    /// Remove the given flag from Self.
+    #[inline]
+    pub fn remove(&mut self, flag: u64) {
+        self.0 &= !(flag);
```

Comment:
Are the parenthesis needed? Is there no warning about it?

```suggestion
        self.0 &= !flag;
```